lib/simulator-xcode-6.js

```diff
@@ -185,6 +185,24 @@ class SimulatorXcode6 extends EventEmitter {
     return appDirs.length !== 0;
   }
 
+  /**
+   * Returns user installed bundle ids which has 'bundleName' in their Info.Plist as 'CFBundleName'
+   * @param {string} bundleId - The bundle id of the application to be checked.
+   */
+  async getUserInstalledBundleIdsByBundleName (bundleName) {
+    const rootUserAppDir = await this.buildBundlePathMap('Bundle');
+    const bundleIds = [];
+    for (const [key, value] of Object.entries(rootUserAppDir)) {
+      const infoPlist = await plist.parsePlistFile(
+        path.resolve(`${await fs.glob(`${value}/*.app`)}`, 'Info.plist'), false);
```

Comment:
does fs.glob always result into something?

Reply:
Yes. I checked some simulator instances. They had below structure when they had apps installed by us.
```
.../Bundle/Application/210A3413-B313-443A-A5EC-CB6CF1C5E84F/IntegrationApp.app/Info.plist (etc)
.../Bundle/Application/210A3413-B313-443A-A5EC-CB6CF1C5E84F/BundleMetadata.plist
```

Reply:
Not a big fan of using the default stringification of a single-entry array. It works for now, but it may not hold that only one `.app` will exist in the directory.